Forbidden Area

wlfrdssn opened the pull request:
Here is the code for the Forbidden Area @perliedman mention in Issues There is still a problem with scaling and I’m not sure how to fix it at the moment. I currently don’t have the time to figure it out. Do you have any suggestions @perliedman? Otherwise, consider implement it as it is and fix the scaling later?

wlfrdssn commented:
There is no function for dashed and solid line for the forbidden area. Maybe a enchancement that could be adressed later.

https://omapwiki.orienteering.sport/symbols/709-out-of-bounds-area-2/

perliedman commented:
Looks like a great start! I have some ideas regarding scaling, but I haven't really done anything like this with OpenLayers before, so it will require some experimentation

perliedman commented:
I added some adjustments for these changes in #44, I think it works well enough to maybe look into merging now. Feel free to try it out and comment.

wlfrdssn commented:
Looks great! I checked it against another map I had with a forbidden area.
When I compared them, I noticed a slight difference in the line thickness.
Otherwise, I didn't see any differences. Merge it! :)

Den lör 15 feb. 2025 kl 16:43 skrev Per Liedman ***@***.***>:

> I added some adjustments for these changes in #44
> <https://github.com/perliedman/o-scout/pull/44>, I think it works well
> enough to maybe look into merging now. Feel free to try it out and comment.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/perliedman/o-scout/pull/37#issuecomment-2660976602>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOVUAX7NYYMEHZIKI3FWKBT2P5OC5AVCNFSM6AAAAABVWWNFEOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRQHE3TMNRQGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: perliedman]*perliedman* left a comment (perliedman/o-scout#37)
> <https://github.com/perliedman/o-scout/pull/37#issuecomment-2660976602>
>
> I added some adjustments for these changes in #44
> <https://github.com/perliedman/o-scout/pull/44>, I think it works well
> enough to maybe look into merging now. Feel free to try it out and comment.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/perliedman/o-scout/pull/37#issuecomment-2660976602>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOVUAX7NYYMEHZIKI3FWKBT2P5OC5AVCNFSM6AAAAABVWWNFEOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRQHE3TMNRQGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

wlfrdssn commented:
I found a problem with the function createForbiddenAreaStyle. When you zoom
out the size will be less than 1. Canvas will treat this as zero and
display an error message. We need to fix it with a fallback, such as "size
= Math.max(size, 1);"

Den mån 17 feb. 2025 kl 22:02 skrev Per Liedman ***@***.***>:

> Merged #37 <https://github.com/perliedman/o-scout/pull/37> into master.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/perliedman/o-scout/pull/37#event-16338875048>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOVUAX2JCJZOQUW4M2MN6L32QJE43AVCNFSM6AAAAABVWWNFEOVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJWGMZTQOBXGUYDIOA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

perliedman commented:
@wlfrdssn good catch, pushed a fix now!
